qiskit_experiments/library/characterization/T2Hahn.py

```diff
@@ -0,0 +1,188 @@
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+T2Hahn Echo Experiment class.
+
+"""
+
+from typing import Union, Iterable, List, Optional
+
+import numpy as np
+from qiskit import QuantumCircuit, QiskitError
+from qiskit.circuit import Measure
+from qiskit.providers.options import Options
+from qiskit.providers import Backend
+
+from qiskit_experiments.framework import BaseExperiment
+
+
+class T2Hahn(BaseExperiment):
+    r"""T2 Ramsey Experiment.
+
+        # section: overview
+
+            This experiment is used to estimate T2 noise of a single qubit.
+
+            See `Qiskit Textbook <https://qiskit.org/textbook/ch-quantum-hardware/\
+            calibrating-qubits-pulse.html>`_  for a more detailed explanation on
+            these properties.
+
+            This experiment consists of a series of circuits of the form
+
+            .. parsed-literal::
+
+                 ┌─────────┐┌──────────┐┌───────┐┌──────────┐┌──────────┐┌─┐
+            q_0: ┤ RY(π/2) ├┤ DELAY(t) ├┤ RX(π) ├┤ DELAY(t) ├┤ RY(-π/2) ├┤M├
+                 └─────────┘└──────────┘└───────┘└──────────┘└──────────┘└╥┘
+            c: 1/═════════════════════════════════════════════════════════╩═
+                                                                         0
+
+            for each *t* from the specified delay times, where
+            :math:`\lambda =2 \pi \times {osc\_freq}`,
+            and the delays are specified by the user.
+            The circuits are run on the device or on a simulator backend.
+
+        # section: tutorial
+            :doc:`/tutorials/t2ramsey_characterization`
+
+        """
+
+    @classmethod
+    def _default_experiment_options(cls) -> Options:
+        """Default experiment options.
+
+        Experiment Options:
+            delays (Iterable[float]): Delay times of the experiments.
+            unit (str): Unit of the delay times. Supported units are
+                's'.
+            n_echos (int); Number of echoes to preform.
+            phase_alt_echo (bool): If to use alternate echoes (must have n_echoes greater than 1)
+        """
+        options = super()._default_experiment_options()
+
+        options.delays = None
+        options.unit = "s"
+        options.n_echos = 1
+        options.phase_alt_echo = False
+
+        return options
+
+    def __init__(
+        self,
+        qubit: Union[int, Iterable[int]],
+        delays: Union[List[float], np.array],  # need to change name?
+        n_echos: int = 1,
+        phase_alt_echo: bool = False,
+    ):
+        """
+        **T2 - Hahn Echo class**
+        Initialize the T2 - Hahn Echo class
+         Args:
```

Comment:
You have an extra space in the beginning from now until the end of the doc string

Reply:
Fixed the amount of space